A6.md

```diff
@@ -0,0 +1,398 @@
+
+Retry policy capabilities are as follows. Each has a detailed description below.
+* [Maximum number of retry attempts](#maximum-number-of-retries)
+* [Exponential backoff](#exponential-backoff)
+* [Set of retryable status codes](#retryable-status-codes)
+
+The hedging policy has the following parameters. See details [here](#hedging-policy).
+* Maximum number of hedged requests
+* Delay between hedged requests
+* Set of non-fatal status codes
+
+Additionally, gRPC provides a mechanism to throttle retry attempts and hedged RPCs when the ratio of failures to successes exceeds a threshold. See [detailed description of throttling](#throttling-retry-attempts-and-hedged-rpcs).
+
+We also provide a mechanism for servers to explicitly signal clients to retry after a settable delay. See [detailed description of server pushback](#pushback).
+
+In some cases, gRPC can guarantee that a request has never been seen by the server application logic. These cases will be transparently retried by gRPC, as detailed [here](#transparent-retries).
+
+Lastly, information about number of retry attempts will be exposed to the client and server applications through metadata. Find more details [here](#exposed-retry-metadata).
+
+### Detailed Design
+
+#### Retry Policy Capabilities
+
+Retry policies support configuring the maximum number of retries, the parameters for exponential backoff, and the set of retryable status codes. Each of these configuration options is detailed in its own section below.
+
+##### Maximum Number of Retries
+
+The maximum number of retry attempts per original RPC. This can be specified in the retry policy as follows:
+
+```
+'max_retry_attempts': 3
+```
+
+gRPC's call deadlines apply to the entire chain of original RPC and retry attempts. The time limit begins when the original RPC is issued to the gRPC client library. For example, if the specified deadline is 500ms and after 500ms only one retry has been attempted, the call is cancelled even if `max_retry_attempts` is set to two or greater.
+
+##### Exponential Backoff
+
+The retry policy may specify the following parameters for exponential backoff:
+
+```
+'exponential_backoff' {
+  'initial_backoff_ms': 1000,
+  'max_backoff_ms': 5000,
+  'multiplier': 3
+}
+```
+
+A value called `current_backoff` is initially set to `initial_backoff`. After every failed RPC, `current_backoff` is set to `min(current_backoff * multiplier, max_backoff)`. The failed RPCs will be retried after *x* seconds, where *x* is defined as `random(0, current_backoff)`.
+
+##### Retryable Status Codes
+
+When gRPC receives a non-OK response status from a server, this status is checked against the set of retryable status codes to determine if a retry attempt should be made.
+
+```
+'retryable_status_codes': {UNAVAILABLE}
```

Comment:
For consistency, let's make all examples in this doc be the exact JSON format that will be used in the service config.